app/src/main/java/com/codesoom/assignment/DemoHttpHandler.java

```diff
@@ -0,0 +1,207 @@
+package com.codesoom.assignment;
+
+import com.codesoom.assignment.domain.Task;
+import com.codesoom.assignment.domain.TaskList;
+import com.codesoom.assignment.dto.TaskDto;
+import com.codesoom.assignment.http.HttpMethod;
+import com.codesoom.assignment.http.HttpStatusCode;
+import com.codesoom.assignment.matcher.TaskUriMatcher;
+import com.codesoom.assignment.matcher.UriMatcher;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import java.io.*;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.codesoom.assignment.http.HttpMethod.*;
+import static com.codesoom.assignment.http.HttpStatusCode.*;
+
+public class DemoHttpHandler implements HttpHandler {
+
+    private static final String LINE_BREAK = "\n";
+
+    private static final String SLASH = "/";
+
+    private static final long EMPTY_RESPONSE_LENGTH = 0;
+
+    private static final long NO_CONTENT_RESPONSE_LENGTH = -1;
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private static final UriMatcher uriMatcher = new TaskUriMatcher();
+
+    private final TaskList taskList;
+
+    public DemoHttpHandler() {
+        taskList = new TaskList();
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+
+        String pathUri = exchange.getRequestURI().getPath();
+
+        String[] paths = pathUri.split(SLASH);
+
+        if (!uriMatcher.isValidPath(paths)) {
```

Comment:
((경로들이 올바른 것)이 아니면) 이렇게 읽게 되네요. `경로들이 올바르지 않은 경로면`이라고 읽히면 좋을 것 같은데 그러면 이렇게 하면 되겠네요.

```suggestion
        if (uriMatcher.isInvalidPath(paths)) {
```

Reply:
확실히 말씀해 주신 내용이 읽기 좋을 거 같습니다.

만약에  **isValidPath** 라는 메소드가 이미 존재하고 사용하고 있고
위와 같은 코드가 필요하다면
```Java
// 추가 
public boolean isInvalidPath(final String[] pathSegments) {
     return !isValidPath(pathSegments);
}
```
이런식으로 추가 해서 하는게 좋을 거 같은데 
괜찮을까요??

Reply:
이미 `isValidPath`가 있었군요 ㅎㅎ 좋습니다